packages/core/src/transaction.ts

```diff
@@ -23,7 +23,7 @@ export class Transaction {
 
   private readonly savepoints: Transaction[] = [];
   private readonly options: RequiredBy<TransactionOptions, 'type' | 'isolationLevel' | 'readOnly'>;
-  private readonly parent: Transaction | null;
```

Comment:
there is no limit to how nested a transaction can be (they become savepoints). I think we should add a getter called `rootTransaction` which returns the very top level transaction and use that instead of parent

Parent can remain public

Reply:
Good point, I've added this now.
LMK if the getter function is what you intended.
It just checks if a parent exists; if so, it returns the `parent.rootTransaction` recursively until parent is null where it returns itself.

Reply:
That's exactly it :)